### PR TITLE
os/kernel/task: add task state of each task shown in assert log

### PR DESCRIPTION
**Pull request by neel-samsung**
This patch adds the task state of each task at the task list of the assert log and TRAP.

Assert log result:
```
===========================================================
List of all tasks in the system
===========================================================
                         NAME |   PID |  PRI |    USED / TOTAL STACK | STACK ALLOC ADDR | TCB ADDR | TASK STATE
-------------------------------------------------------------------------------------------------------------------
                    Idle Task |     0 |    0 |     676 /    1024 |       0x10004bfc | 0x100294f0 |          2
                       hpwork |     1 |  201 |     264 /    2028 |       0x1005ecb0 | 0x1005e530 |          7
                       lpwork |     2 |   50 |     264 /    2028 |       0x1005fc40 | 0x1005f4c0 |          7
                  LWIP_TCP/IP |     3 |  105 |     356 /    4068 |       0x1006c1f0 | 0x1006ba70 |          5
         netmgr_event_handler |     4 |  100 |     208 /    2012 |       0x1006d980 | 0x1006d200 |          5
               binary_manager |     5 |  203 |     756 /    8164 |       0x1006e980 | 0x1006e200 |          8
            bm_faultmsgsender |     6 |  205 |     216 /     988 |       0x10071110 | 0x10070990 |          6
                         app1 |     8 |  220 |     408 /    4068 |        0x2100470 | 0x10072e50 |          7
                         app2 |    10 |  180 |     628 /    8164 |        0x2181b20 | 0x100716c0 |          7
                        uwork |    11 |  100 |     288 /    2028 |        0x2184090 | 0x10071b90 |          7
                         tash |    12 |  125 |     756 /    4076 |        0x2184b50 | 0x10072060 |          5
             wifi msg handler |    13 |  100 |     400 /    4060 |        0x21861d0 | 0x10072530 |          5
                        hello |    14 |  100 |     476 /    4076 |        0x2187630 | 0x10072b10 |          3
-------------------------------------------------------------------------------------------------------------------
```
TRAP RESULT:
```
                        NAME |   PID |  PRI |    USED / TOTAL STACK | STACK ALLOC ADDR | TCB ADDR | TASK STATE

-------------------------------------------------------------------------------------------------------------------

                   Idle Task |     0 |    0 |     676 /    1024 |       0x10004bfc | 0x100294f0 |  Wait to scheduling (Ready)

                      hpwork |     1 |  201 |     264 /    2028 |       0x1005ecb0 | 0x1005e530 |  Wait Signal

                      lpwork |     2 |   50 |     264 /    2028 |       0x1005fc40 | 0x1005f4c0 |  Wait Signal

                 LWIP_TCP/IP |     3 |  105 |     356 /    4068 |       0x1006c1f0 | 0x1006ba70 |  Wait Semaphore

        netmgr_event_handler |     4 |  100 |     208 /    2012 |       0x1006d980 | 0x1006d200 |  Wait Semaphore

              binary_manager |     5 |  203 |     756 /    8164 |       0x1006e980 | 0x1006e200 |  Wait MQ Receive (MQ Empty)

           bm_faultmsgsender |     6 |  205 |     216 /     988 |       0x10071110 | 0x10070990 |  Wait FIN

                        app1 |     8 |  220 |     408 /    4068 |        0x2100470 | 0x10072e50 |  Wait Signal

                        app2 |    10 |  180 |     628 /    8164 |        0x2181b20 | 0x100716c0 |  Wait Signal

                       uwork |    11 |  100 |     288 /    2028 |        0x2184090 | 0x10071b90 |  Wait Signal

                        tash |    12 |  125 |     756 /    4076 |        0x2184b50 | 0x10072060 |  Wait Semaphore

            wifi msg handler |    13 |  100 |     400 /    4060 |        0x21861d0 | 0x10072530 |  Wait Semaphore

                       hello |    14 |  100 |     476 /    4076 |        0x2187630 | 0x10072b10 |  Running

-------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Review by sunghan-chang — COMMENTED** [review]
How about the TRAP parsing for task state?
The log will show just a number. User can't know what it means.
Could you change a number to readable message on the TRAP?

**Comment by neel-samsung** [comment]
> How about the TRAP parsing for task state? The log will show just a number. User can't know what it means. Could you change a number to readable message on the TRAP?



> How about the TRAP parsing for task state? The log will show just a number. User can't know what it means. Could you change a number to readable message on the TRAP?

yes, We can parse that and turn it into meaningful message. can we do it like adding task state and also its desc instead of task state number?
```
                         NAME |   PID |  PRI |    USED / TOTAL STACK | STACK ALLOC ADDR | TCB ADDR | TASK STATE
-------------------------------------------------------------------------------------------------------------------
                    Idle Task |     0 |    0 |     664 /    1024 |       0x10004bfc | 0x100294f0 |          2
                       hpwork |     1 |  201 |     264 /    2028 |       0x1005ecb0 | 0x1005e530 |          5
                       lpwork |     2 |   50 |     264 /    2028 |       0x1005fc40 | 0x1005f4c0 |          7
-------------------------------------------------------------------------------------------------------------------
```
My suggestion
```
                         NAME |   PID |  PRI |    USED / TOTAL STACK | STACK ALLOC ADDR | TCB ADDR | TASK STATE
-------------------------------------------------------------------------------------------------------------------
                    Idle Task |     0 |    0 |     664 /    1024 |       0x10004bfc | 0x100294f0 |   READY-TO-RUN - But not running
                       hpwork |     1 |  201 |     264 /    2028 |       0x1005ecb0 | 0x1005e530 |   BLOCKED      - Initialized but not yet activated 
                       lpwork |     2 |   50 |     264 /    2028 |       0x1005fc40 | 0x1005f4c0 |   BLOCKED      - Waiting for a signal
-------------------------------------------------------------------------------------------------------------------
```

**Comment by sunghan-chang** [comment]
@neel-samsung I think we can do this on the TRAP only, not on the assert log.
In message, how about below?  
1. Running
2. Wait to scheduling (Ready)
3. Pending preemption unlock
4. Wait Signal
5. Wait MQ Receive (MQ Empty)
6. Wait MQ Send (MQ Full)
7. Wait FIN
8. Inactive
??

**Comment by neel-samsung** [comment]
> @neel-samsung I think we can do this on the TRAP only, not on the assert log. In message, how about below?
> 
> 1. Running
> 2. Wait to scheduling (Ready)
> 3. Pending preemption unlock
> 4. Wait Signal
> 5. Wait MQ Receive (MQ Empty)
> 6. Wait MQ Send (MQ Full)
> 7. Wait FIN
> 8. Inactive
> 9. else Invalid state
>    ??
Thank you for the suggestion. Yes, we can do this on TRAP only.

**Review by samsung-singh — APPROVED** [review]
kernel task adding task state of each task shown in assert log, good to merge.

File: tools/trap/ramdumpParser.py [hunk 1]
```diff
@@ -1078,7 +1078,29 @@ def parseCorruptHeapInfo(line):
...
+                    "1" : " Pending preemption unlock",
+                    "2" : " Wait to scheduling (Ready)",
+                    "3" : " Running",
+                    "4" : " Inactive",
+                    "5" : " Wait Semaphore",
+                    "6" : " Wait FIN",
+                    "7" : " Wait Signal",
+                    "8" : " Wait MQ Receive (MQ Empty)",
+                    "9" : " Wait MQ Send (MQ Full)",
+				}
```
Comment (amogh-samsung) [inline tools/trap/ramdumpParser.py:1092]: Minor, indentation?

**Review by amogh-samsung — APPROVED** [review]

**Comment by sunghan-chang** [comment]
@neel-samsung Could you leave a trap result with task state in the first commit description?

**Comment by neel-samsung** [comment]
> @neel-samsung Could you leave a trap result with task state in the first commit description? You can leave it with task list information only which shows a task state. (I mean that you don't need to leave entire result, you can leave changed part only.)

I have added it in a PR description. Please check and review.

**Review by gSahitya-samsung — APPROVED** [review]

**Comment by sunghan-chang** [comment]
> I have added it in a PR description. Please check and review.

@neel-samsung Let's add it in the commit description. I don't want to do it in the PR description only. You don't need to do in the PR if you do in the commit.
```
@neel-samsung Could you leave a trap result with task state in the first commit description?
```
This is what I left.

**Comment by sunghan-chang** [comment]
@neel-samsung When you check the repo changes later, you will use a commit history, you will not do in the pull request tab. That's why we should write details well in a commit.

**Comment by sunghan-chang** [comment]
I never ask to modify pr title or description.

**Comment by neel-samsung** [comment]
> > I have added it in a PR description. Please check and review.
> 
> @neel-samsung Let's add it in the commit description. I don't want to do it in the PR description only. You don't need to do in the PR if you do in the commit.
> 
> ```
> @neel-samsung Could you leave a trap result with task state in the first commit description?
> ```
> 
> This is what I left.

@sunghan-chang I am very sorry, we will add it in commit description.

**Comment by neel-samsung** [comment]
> @neel-samsung Could you leave a trap result with task state in the first commit description? You can leave it with task list information only which shows a task state. (I mean that you don't need to leave entire result, you can leave changed part only.)

@sunghan-chang I have added commit description. Please check and review.

**Review by r-prabu — APPROVED** [review]

**Review by gSahitya-samsung — APPROVED** [review]

File: os/kernel/task/task_dumptasks.c [hunk 2]
```diff
@@ -36,27 +36,27 @@
 
 #ifdef CONFIG_STACK_COLORATION
 #if CONFIG_TASK_NAME_SIZE > 0
-#define TASKDUMP_ARGS_FORMAT "%*s | %5s | %4s | %7s / %7s | %16s | %8s\n"
-#define TASKDUMP_FORMAT "%*s | %5d | %4d | %7lu / %7lu | %16p | %8p\n"
-#define TASKDUMP_VALUE  CONFIG_TASK_NAME_SIZE, tcb->name, tcb->pid, tcb->sched_priority, (unsigned long)used_stack_size, (unsigned long)tcb->adj_stack_size, tcb->stack_alloc_ptr, tcb
-#define TASKDUMP_ARGS  CONFIG_TASK_NAME_SIZE, "NAME", "PID", "PRI", "USED", "TOTAL STACK",  "STACK ALLOC ADDR", "TCB ADDR"
+#define TASKDUMP_ARGS_FORMAT "%*s | %5s | %4s | %7s / %7s | %16s | %8s | %10s\n"
+#define TASKDUMP_FORMAT "%*s | %5d | %4d | %7lu / %7lu | %16p | %8p | %10u\n"
+#define TASKDUMP_VALUE  CONFIG_TASK_NAME_SIZE, tcb->name, tcb->pid, tcb->sched_priority, (unsigned long)used_stack_size, (unsigned long)tcb->adj_stack_size, tcb->stack_alloc_ptr, tcb, tcb->task_state
```
Comment (gSahitya-samsung) [inline os/kernel/task/task_dumptasks.c:41]: test

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**